### PR TITLE
src/circle.js: Improve the readability of the point(x, y)

### PR DESCRIPTION
**Pull request by martinfrances107**
I am slowly converting this project to rust .. and so I am crawling over the code base function by function.

I don't want to change to implementation, just provide equivalent code the is easier to read for the next developer.

How to interpret point(x,y). Let me put it into conversational English 

1) Use the rotate transform to manipulate the input.
2) Convert the scalar variable x into a array that contains the conversion.
3) push this new x  onto the ring.
4) Retroactively go back and convert what has been pushed into radians

The more natural flow is to not reuse x and convert to radians before pushing.

Anyway, I hope readability improvements are considered an acceptable change.

### CODE REVIEW

**Comment by Fil** [comment]
It would need a speed & memory analysis, because this is one of the innermost loops and it should be as fast as possible, even if the price is a slightly lower readability.

In general d3-geo is hard to read and the community would benefit, in the long run, from a heavily commented version (maybe not in the code itself but as a series of "literate programming" notebooks, explaining each function's algorithm and quirks?). This is something I'd really love to put some time in.

**Comment by martinfrances107** [comment]
> It would need a speed & memory analysis, because this is one of the innermost loops and it should be as fast as possible, even if the price is a slightly lower readability.

I completely understand 

I can see from my porting of  this code to rust .. that is going to be a critical loop.

The way I am working on the port is

STEP A -  get an initial  skeleton version going which passes all the associated ported test
STEP B - refactor the rust using a flame graph and other aspects of RUST profiling eco-system.  

Regarding d3-geo-voronoi, which I see you wrote... :)   and a  project that makes heavy use of this project  :)

I am aiming for a millestone  early next week where I have  Step A complete for voronoi.js and delaunay.js
[ I already have step A complete for the subset of d3-geo that is invoked  by d3-geo-voronoi ]
[ The subject is projection, rotation, length.js and distance.js ] 

In short once I hit that milestone and move to STEP B -- which will involve indirectly profiling the javascript we are talking about here .. Then I will be in a position to comment about how much this change increases or decreases the new test.

 I want this to work now .. but in practical terms I hope to comment here with news in about a month.

**Comment by jrus** [comment]
I would expect this to have the same speed. It’s doing the same operations. Personally I don’t find it any more or less legible though.

**Comment by martinfrances107** [comment]
> If you want to speed it up, the thing to worry about would be the rotateRadians(...).invert function, which takes 2 calls to cos, 2 calls to sin, 1 call to atan2, 1 call to asin, and creates a new JS array. That part is going to dominate the computation time here.

Thanks for putting in that observation .... as I profile the RUST port I will look at that.

As far a getting consensus on this change .. I see that I have failed.

It is the 

Retroactively go back and convert what has been pushed into radians

that bugs me, but hey I have made my case ... and I realise other software engineers have a different focus and different values ... in short I would be OK if this issue is closed.
